qtism/common/enums/AbstractEnumeration.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+namespace qtism\common\enums;
+
+use InvalidArgumentException;
+
+/**
+ * Abstract enumeration automatizes the translation from constant name to
+ * value and from value to name.
+ */
+abstract class AbstractEnumeration implements Enumeration
+{
+    abstract public static function asArray(): array;
+
+    public static function getConstantByName($name)
+    {
+        return static::asArray()[$name] ?? false;
+    }
+
+    public static function getNameByConstant($constant)
```

Comment:
can we add return typehint here?

Reply:
no because we do not have such type string|false